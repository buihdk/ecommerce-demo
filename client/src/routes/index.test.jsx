import React, { Suspense } from 'react';
import { act } from 'react-dom/test-utils';
import { MemoryRouter } from 'react-router';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Routes from '.';

jest.spyOn(require('../services'), 'fetchItems').mockResolvedValue(
  new Promise(resolve => {
    resolve({ data: [] });
  }),
);
jest.spyOn(require('../services'), 'fetchItemById').mockResolvedValue(
  new Promise(resolve => {
    resolve({ category_id: 1 });
  }),
);

describe('Routes', () => {
  let wrapper;

  test('renders ItemListPage without crashing', async () => {
    await act(async () => {
      wrapper = await mount(
        <MemoryRouter initialEntries={['/']}>
          <Suspense fallback={<div>loading...</div>}>
            <Routes />
          </Suspense>
        </MemoryRouter>,
      );
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('renders ItemDetailPage without crashing', async () => {
    await act(async () => {
      wrapper = await mount(
        <MemoryRouter initialEntries={['/items/1']}>
          <Suspense fallback={<div>loading...</div>}>
            <Routes />
          </Suspense>
        </MemoryRouter>,
      );
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('renders NotFoundPage without crashing', async () => {
    await act(async () => {
      wrapper = await mount(
        <MemoryRouter initialEntries={['/404']}>
          <Suspense fallback={<div>loading...</div>}>
            <Routes />
          </Suspense>
        </MemoryRouter>,
      );
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
