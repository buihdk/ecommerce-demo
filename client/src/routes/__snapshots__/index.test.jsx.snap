// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes renders ItemDetailPage without crashing 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/items/1",
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "1nme7j",
            "pathname": "/items/1",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "1nme7j",
          "pathname": "/items/1",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Suspense
      fallback={
        <div>
          loading...
        </div>
      }
    >
      <Memo(Routes)>
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {
                  "id": "1",
                },
                "path": "/items/:id",
                "url": "/items/1",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "key": "1nme7j",
                "pathname": "/items/1",
                "search": "",
                "state": undefined,
              }
            }
            path="/items/:id"
          >
            <component
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "1nme7j",
                      "pathname": "/items/1",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "1nme7j",
                    "pathname": "/items/1",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "1nme7j",
                  "pathname": "/items/1",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {
                    "id": "1",
                  },
                  "path": "/items/:id",
                  "url": "/items/1",
                }
              }
            />
          </Route>
        </Switch>
      </Memo(Routes)>
      <div>
        loading...
      </div>
    </Suspense>
  </Router>
</MemoryRouter>
`;

exports[`Routes renders ItemListPage without crashing 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "d8gvcg",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "d8gvcg",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Suspense
      fallback={
        <div>
          loading...
        </div>
      }
    >
      <Memo(Routes)>
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "key": "d8gvcg",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <component
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "d8gvcg",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "d8gvcg",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "d8gvcg",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            />
          </Route>
        </Switch>
      </Memo(Routes)>
      <div>
        loading...
      </div>
    </Suspense>
  </Router>
</MemoryRouter>
`;

exports[`Routes renders NotFoundPage without crashing 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/404",
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "opt11y",
            "pathname": "/404",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "opt11y",
          "pathname": "/404",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Suspense
      fallback={
        <div>
          loading...
        </div>
      }
    >
      <Memo(Routes)>
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {
                  "0": "/404",
                },
                "path": "*",
                "url": "/404",
              }
            }
            location={
              Object {
                "hash": "",
                "key": "opt11y",
                "pathname": "/404",
                "search": "",
                "state": undefined,
              }
            }
            path="*"
          >
            <component
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "opt11y",
                      "pathname": "/404",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "opt11y",
                    "pathname": "/404",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "opt11y",
                  "pathname": "/404",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {
                    "0": "/404",
                  },
                  "path": "*",
                  "url": "/404",
                }
              }
            />
          </Route>
        </Switch>
      </Memo(Routes)>
      <div>
        loading...
      </div>
    </Suspense>
  </Router>
</MemoryRouter>
`;
